### PR TITLE
update to new opendbc api

### PR DESCRIPTION
**Pull request by deanlee**
depends on https://github.com/commaai/opendbc/pull/934

performance is twice as slow because of a few more data conversions. but it should be enough.

the results of running `selfdrive/debug/check_can_parser_performance.py` :
[updated]:
before:
> 125.61 mean ms, 130.85 max ms, 118.84 min ms, 2.78 std ms

after:
>  263.47 mean ms, 367.93 max ms, 258.64 min ms, 37.38 std ms

### CODE REVIEW

File: selfdrive/car/interfaces.py [hunk 1]
```diff
@@ -217,9 +218,10 @@ def _update(self, c: car.CarControl) -> car.CarState:
 
   def update(self, c: car.CarControl, can_strings: list[bytes]) -> car.CarState:
     # parse can
+    can_list = can_capnp_to_list(can_strings)
```
Comment (deanlee) [inline selfdrive/car/interfaces.py:221]: because only one conversion (capnp->list) needs to be done.  the overall performance will be better than the previous version of opendbc.

Comment (adeebshihadeh) [reply selfdrive/car/interfaces.py:221]: so it's not twice as slow?

Comment (deanlee) [reply selfdrive/car/interfaces.py:221]: updated the test case to remove the overhead of conversion: https://github.com/commaai/openpilot/pull/32048
the new test results is:

before:

> 343.58 mean ms, 349.11 max ms, 339.32 min ms, 3.40 std ms

after:

> 476.56 mean ms, 498.29 max ms, 464.91 min ms, 10.96 std ms

It's still a little slower, but not twice as much.

Comment (adeebshihadeh) [reply selfdrive/car/interfaces.py:221]: hmm, even the old speed seems a bit slow. any other optimizations we can do at least get it to be the same?

Comment (deanlee) [reply selfdrive/car/interfaces.py:221]: I'll thinking of ways to optimize further.

Comment (deanlee) [reply selfdrive/car/interfaces.py:221]: Submitted a PR https://github.com/commaai/opendbc/pull/1039 to improve the performance of `update_string`, now it's twice as fast. This PR will also benefit from it once merged.

**Comment by sshane** [comment]
@deanlee this look good to you for the benchmark? https://github.com/commaai/openpilot/pull/32009/commits/b494b1442e51db25c7e8b6deffcae1149b3ea67b

File: selfdrive/debug/check_can_parser_performance.py [hunk 2]
```diff
@@ -25,12 +26,12 @@ class CarModelTestCase(TestCarModelBase):
   CC = car.CarControl.new_message()
   ets = []
   for _ in tqdm(range(N_RUNS)):
-    msgs = [(m.as_builder().to_bytes(),) for m in tm.can_msgs]
+    msgs = [m.as_builder().to_bytes() for m in tm.can_msgs]
     start_t = time.process_time_ns()
     for msg in msgs:
       for cp in tm.CI.can_parsers:
```
Comment (sshane) [inline selfdrive/debug/check_can_parser_performance.py:32]:
```suggestion
      for cp in tm.CI.can_parsers:
      can_list = can_capnp_to_list([msg])
```

**Comment by sshane** [comment]
@deanlee panda shouldn't be touched: https://github.com/commaai/openpilot/pull/32009/commits/17b76ca0cc66cf36d8db15835f2b1bfb13fc6436

**Comment by sshane** [comment]
trigger-jenkins

**Comment by sshane** [comment]
Seeing same CPU usage in card which is great!

**Comment by deanlee** [comment]
Great to hear the CPU usage is the same! We can further reduce it later, like in pr https://github.com/commaai/opendbc/pull/1046

**Comment by sshane** [comment]
trigger-jenkins

File: selfdrive/controls/lib/longitudinal_mpc_lib/SConscript [hunk 3]
```diff
@@ -66,7 +66,7 @@ lenv.Clean(generated_files, Dir(gen))
 generated_long = lenv.Command(generated_files,
                               source_list,
                               f"cd {Dir('.').abspath} && python3 long_mpc.py")
-lenv.Depends(generated_long, [msgq_python, common_python, opendbc_python])
+lenv.Depends(generated_long, [msgq_python, common_python, opendbc_python, pandad_python])
```
Comment (sshane) [inline selfdrive/controls/lib/longitudinal_mpc_lib/SConscript:69]: Why is this needed?

Comment (deanlee) [reply selfdrive/controls/lib/longitudinal_mpc_lib/SConscript:69]: Forgot why. Maybe because I added pandad_python to all modules that depend on opendbc_python. It should not be needed here.

Comment (sshane) [reply selfdrive/controls/lib/longitudinal_mpc_lib/SConscript:69]: I think it may be because radard uses it, and the MPC imports something from radard.

Comment (deanlee) [reply selfdrive/controls/lib/longitudinal_mpc_lib/SConscript:69]: update: remove them will cause compile errors:

  File "/home/dean/Projects/openpilot/selfdrive/controls/lib/longitudinal_mpc_lib/long_mpc.py", line 11, in <module>
    from openpilot.selfdrive.car.interfaces import ACCEL_MIN
  File "/home/dean/Projects/openpilot/openpilot/selfdrive/car/interfaces.py", line 21, in <module>
    from openpilot.selfdrive.pandad import can_capnp_to_list
  File "/home/dean/Projects/openpilot/openpilot/selfdrive/pandad/__init__.py", line 2, in <module>
    from openpilot.selfdrive.pandad.pandad_api_impl import can_list_to_can_capnp, can_capnp_to_list
ModuleNotFoundError: No module named 'openpilot.selfdrive.pandad.pandad_api_impl'

Comment (sshane) [reply selfdrive/controls/lib/longitudinal_mpc_lib/SConscript:69]: I tried this, but realized long_mpc also imports from interfaces, so your approach is the simplest

```
class RadarInterfaceBase(ABC):
  def __init__(self, CP):
    self.rcp = None
    self.pts = {}
    self.delay = 0
    self.radar_ts = CP.radarTimeStep
    self.frame = 0

  def update(self, can_strings: list[bytes]):
    return self._update(can_capnp_to_list(can_strings))

  @abstractmethod
  def _update(self, can_list):
    pass

  def no_radar_response(self):
    self.frame += 1
    if (self.frame % int(100 * self.radar_ts)) == 0:
      return car.RadarData.new_message()
    return None
```

**Comment by sshane** [comment]
trigger-jenkins
